### PR TITLE
Spray Bottle and Fire Extinguisher Fixes

### LINKED ISSUE #5845: Fire Extinguisher Sprays Are Not Colored To Reflect Reagents Inside
Spraying polyacid from a fire extinguisher looks no different to observers than spraying water. Foam extinguishers, on the other hand, spray purple foam when filled with polyacid.

### LINKED ISSUE #5864: (WEB REPORT BY: johnnychimp REMOTE: 198.245.63.50:7777) Spray bottles spray black spritz when empty
>Revision (Should be above if you're viewing this from ingame!)

>General description of the issue
When a pepper spray or space cleaner bottle is empty, you can still spray but it is a wierd black spritz.

>What you expected to happen

>What actually happened

>Steps to reproduce if possible

Empty a spray bottle and spray it.

### PR DESCRIPTION
**Pull request by Shadowmech88**
Fixes #5864.
Fixes #5845.

As a result of fire extinguisher sprays now reflecting their reagent color, the default water extinguisher sprays are no longer a deep blue color, due to the water reagent itself not actually being deep blue in the game.
water in a beaker

### CODE REVIEW

File: code/game/objects/effects/effect_system.dm [hunk 1]
```diff
@@ -29,6 +33,17 @@ would spawn and follow the beaker, even if it is carried or thrown.
 	spawn(70)
 		qdel(src)
 
+/obj/effect/effect/water/spray/New(loc, datum/reagents/R)
+	reagents = R
+	reagents.my_atom = src
+	var/ccolor = mix_color_from_reagents(reagents.reagent_list)
+	if(ccolor)
+		icon += ccolor
```
Comment (RemieRichards) [inline code/game/objects/effects/effect_system.dm]: probably nicer to set the spray's ```color``` var to ccolor.

File: code/game/objects/effects/effect_system.dm [hunk 2]
```diff
@@ -29,6 +33,17 @@ would spawn and follow the beaker, even if it is carried or thrown.
 	spawn(70)
 		qdel(src)
 
+/obj/effect/effect/water/spray/New(loc, datum/reagents/R)
```
Comment (clusterfack) [inline code/game/objects/effects/effect_system.dm]: Why in the world are you doing this for each spray instead of using the shot

**Comment by clusterfack** [comment]
Check this for foam extinguishers also

**Comment by Shadowmech88** [comment]
The color? Foam extinguishers already reflect their reagent content in the color of their foam.
That's why it was so odd that regular extinguishers didn't do it.

**Comment by Shadowmech88** [comment]
Did not mean to close this.

**Comment by clusterfack** [comment]
Well who made water not blue anymore? Pretty sure I remember a time when it was

**Comment by Probe1** [comment]
:+1: 

also fix foam extinguishers firing pure white tiles of foam

**Comment by Shadowmech88** [comment]
@Probe1 
As I mentioned, foam extinguishers already account for reagent color.
polyacid foam
welding fuel foam
The tiles looking solid white when it sprays water is a fault of the water's reagent color, not of the extinguisher itself.

**Comment by clusterfack** [comment]
Thus someone changed waters reagent color from blue, the monster

**Pull request merged**